src/PhpParser/FileParser.php

```diff
@@ -69,4 +70,14 @@ public function parse(SplFileInfo $fileInfo): array
             throw UnparsableFile::fromInvalidFile($filePath, $throwable);
         }
     }
+
+    /**
+     * Return tokens for the last parse.
+     *
+     * @return Token[]
+     */
+    public function getTokens(): array
```

Comment:
I'm not a fan of this API. It fits in the current usage but it lacks any cohesiveness. Can't we change `FileParser` to return a tuple instead?

```suggestion
    /**
     * @throws UnparsableFile
     *
     * @return array{Stmt[], Token[]}
     */
    public function parse(SplFileInfo $fileInfo): array
    {
        try {
            $statements = $this->parser->parse($fileInfo->getContents()) ?? [];
            $tokens = $this->parser->getTokens();

            return [$statements, $tokens];
        } catch (Throwable $throwable) {
            $filePath = $fileInfo->getRealPath() === false
                ? $fileInfo->getPathname()
                : $fileInfo->getRealPath()
            ;

            throw UnparsableFile::fromInvalidFile($filePath, $throwable);
        }
    }
```

I appreciate it has a greater impact though

Reply:
Decided not to do it because `->parse($file)` and `->getTokens()` this is exactly the API that `php-parser` exposes. Yes, it's arguable, but we only have a wrapper.

Such change will affect many files and IMO if we really want it (I have doubts) it shouldn't be done here.